configure.ac

```diff
@@ -593,6 +598,21 @@ AS_CASE([$ocaml_cv_cc_vendor],
 ocamlsrcdir=$(unset CDPATH; cd -- "$srcdir" && printf %sX "$PWD") || fail
 ocamlsrcdir=${ocamlsrcdir%X}
 
+# Whether response files are supported
+
+AS_CASE([$enable_ar_diversion],
+  [yes], [ar_supports_diversion=true],
+  [no], [ar_supports_diversion=false],
+  [AS_CASE([$host],
+    [*-pc-windows],
+      [ar_supports_diversion=true],
+    [AC_MSG_CHECKING([whether ar supports @file arguments])
+    AS_IF([$AR @/dev/null --help >/dev/null 2>&1],
```

Comment:
`ar` on macOS doesn't support the `--help` option, so you're testing for whether `ar` supports diversions AND the `--help` option...